WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -194,7 +194,7 @@ private async Task<ImmutableArray<AliceClient>> CreateRegisterAndConfirmCoinsAsy
 					var identificationMasterKey = Slip21Node.FromSeed(masterKey.ToBytes());
 					var identificationKey = identificationMasterKey.DeriveChild("SLIP-0019").DeriveChild("Ownership identification key").Key;
 
-					return await AliceClient.CreateRegisterAndConfirmInputAsync(roundState, aliceArenaClient, coin, secret, identificationKey, RoundStatusUpdater, cancellationToken).ConfigureAwait(false);
+					return await AliceClient.CreateRegisterAndConfirmInputAsync(personCircuit, roundState, aliceArenaClient, coin, secret, identificationKey, RoundStatusUpdater, cancellationToken).ConfigureAwait(false);
```

Comment:
This is not okay because `AliceClient` doesn't have to know about communication layer nor any other plumbing concept. All the `AliceClient` needs to know is that it has access to a component that can handle (in a transparent way) all its requests: the `IWabiSabiApiRequestHandler`.

In this code the `arenaRequestHandler` (an `IWabiSabiApiRequestHandler`) needs to know the circuit and the change in L178-L179 are okay but `AliceClient` should know nothing. 

Btw, each `AliceClient` has its own and exclusive instance of `IWabiSabiApiRequestHandler` and then lifetime of both are the same. In this particular case, each request handler should have its own and exclusive personCircuit and then the three components `AliceClient`, `IWabiSabiApiRequestHandler` and `PersonCircuit` should have the same lifetime.